tests/v2/validation/publicapi/clusterroletemplatebindings/crtb_create_steve_test.go

```diff
@@ -0,0 +1,307 @@
+package clusterroletemplatebindings
+
+import (
+	"encoding/json"
+	"testing"
+
+	v1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type CRTBGenTestSuite struct {
+	suite.Suite
+	client            *rancher.Client
+	crtbName          string
+	roleTemplateName  string
+	userPrincipalName string
+	session           *session.Session
+	cluster           *management.Cluster
+	clusterName       string
+	remoteClusterName string
+	clusterID         string
+	remoteClusterID   string
+	clusterV1         *v1.Cluster
+	jsonData          []byte
+}
+
+const (
+	crtbJSONFilePath        = "resources/createCRTBTemplate.json"
+	crtbApiEndPoint         = "management.cattle.io.clusterroletemplatebinding"
+	defaultRoleTemplateName = "cluster-member"
+)
+
+func (crtb *CRTBGenTestSuite) TearDownSuite() {
+	crtb.session.Cleanup()
+}
+
+func (crtb *CRTBGenTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	crtb.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(crtb.T(), err)
+
+	crtb.client = client
+	crtb.clusterName = client.RancherConfig.ClusterName
+	require.NotEmptyf(crtb.T(), crtb.clusterName, "Cluster name to install should be set")
+
+	crtb.clusterID, err = clusters.GetClusterIDByName(crtb.client, crtb.clusterName)
+	require.NoError(crtb.T(), err, "Error getting cluster ID")
+	crtb.cluster, err = crtb.client.Management.Cluster.ByID(crtb.clusterID)
+	require.NoError(crtb.T(), err)
+
+	crtb.remoteClusterName = getNonLocalCluster(crtb.T(), client)
+	crtb.remoteClusterID, err = clusters.GetClusterIDByName(crtb.client, crtb.remoteClusterName)
+	require.NoError(crtb.T(), err)
+
+	crtb.jsonData = createJsonTemplateForClusterRoleTemplateBinding().jsonData
+	require.NoError(crtb.T(), err)
+}
+
+func getNonLocalCluster(t *testing.T, client *rancher.Client) string {
```

Comment:
This could be implemented as an extension. FYI - I already have a similar function in the clusters extension for listing the downstream clusters in my PR here: https://github.com/rancher/rancher/pull/43317/commits/2d4f463c1b294ff17c35db1ae762d636c2d5e362#diff-ec8a0f11840abbf14048e4eb830f81995928f1322ebd73393e4d68186e310c4b.

Reply:
I removed this function for now, i actually added this function to implement kubectl on a local cluster and create crtbs on a downstream cluster. I created this function in advance to use it on both places. To reduce the confusion, i deleted this function for now and using the regular flow.